lookout/style/format/analyzer.py

```diff
@@ -67,19 +83,35 @@ def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
                     comment.confidence = 100
                     comment.line = 1
                     comment.text = "Failed to parse this file"
+                    comments.append(comment)
                     continue
                 X, y, vnodes = res
                 X, _ = fe.select_features(X, y)
-                self.log.debug("predicting values for %d samples", len(y))
-                y_pred, winners = rules.predict(X, True)
+                feature_names = fe.selected_feature_names
+                y_pred, rule_winners = rules.predict(X, True)
                 assert len(y) == len(y_pred)
 
-                for yi, y_predi, vnode, winner in zip(y, y_pred, vnodes, winners):
-                    if yi != y_predi:
-                        comment = vnode.to_comment(y_predi)
-                        comment.file = file.path
-                        comment.confidence = int(round(rules.rules[winner].stats.conf * 100))
-                        comments.append(comment)
+                if self.config["debug"]:
+                    code_lines = file.content.decode("utf-8", "replace").splitlines()
+                for line_nodes in one_line_nodes(y, y_pred, vnodes, rule_winners):
+                    code_line_number = line_nodes[0][2].start.line - 1
+                    code_text = (
+                        "" if not self.config["debug"] else
+                        "```%s\n%s\n```\n\n" % (lang, code_lines[code_line_number]))
+                    vnodes_comments = ["%s\n%s\n" % (
+                        vnode.to_comment(y_predi),
```

Comment:
We should rename `to_comment` because the new function does not really produce a comment - it produces a string which we use as a comment, but don't have to.

Reply:
I would suggest to remove it completely and define one here.

Reply:
In the future we will try to make it sexy and move all the code here inside the Rule